src/main/java/com/google/songgame/servlets/GameServlet.java

```diff
@@ -159,13 +167,47 @@ public static YouTube getService() throws GeneralSecurityException, IOException
   }
 
   /**
-   * Returns video ID
+   * Retrieve random video from array of videos and stores video in datastore
    */
   private String getRandomVideo(ArrayList<String> playlistVideos) {
     Random randomGenerator = new Random();
     int playlistSize = playlistVideos.size();
     int index = randomGenerator.nextInt(playlistSize);
     String videoId = playlistVideos.get(index);
+    
+    try {
+      Video currentVideo = getVideoInfo(videoId);
+      setVideoInfo(currentVideo);
```

Comment:
Generally you want to avoid having "side effects" like this inside getter methods, since it's a lot easier to reason about what getter methods are doing if they only return things and don't change application state. For example, you wouldn't be able to tell from looking at line 88 of this class that the video is being stored anywhere.

Instead, it would be better to move this block of code into its own "storeVideoInfo(String videoId)" which is called immediately after line 88.